packages/webhooks/model-validator/validator.js

```diff
@@ -0,0 +1,208 @@
+            throw new Error('Schema file type was not supported. For now only .ts files are supported.')
+        } else if (!fs.existsSync(schemaPath)) {
+            throw new Error('Schema file does not exist!')
+        }
+        const program = tjs.getProgramFromFiles([schemaPath])
+        const settings = {
+            noExtraProps: true,
+        }
+        this.generator = tjs.buildGenerator(program, settings)
+        this.models = []
+        this.filterValidators = {}
+        this.fieldsValidators = {}
+    }
+
+    /**
+     * Registers GraphQL model from schema in internal base
+     * Also generates JSON-schema validators for validating its fields and filters
+     * @param {string} modelName
+     */
+    registerModel (modelName) {
+        if (this.models.includes(modelName)) return
+
+        this.models.push(modelName)
+
+        const modelFilterSchema = this.generator.getSchemaForSymbol(`${modelName}WhereInput`)
+        const ajv = new Ajv({ strict: true, allErrors: true })
+        ajv.validateSchema(modelFilterSchema)
+        if (ajv.errors) {
+            throw new Error(`Schema for class ${modelName}WhereInput was not valid!`)
+        }
+        this.filterValidators[modelName] = ajv.compile(modelFilterSchema)
+
+        const modelSchema = this.generator.getSchemaForSymbol(modelName)
+        ajv.validateSchema(modelSchema)
+        if (ajv.errors) {
+            throw new Error(`Schema for class ${modelName} was not valid!`)
+        }
+        const maskedSchema = WebHookModelValidator._maskSchemaTypes(modelSchema)
+        ajv.validateSchema(maskedSchema)
+        if (ajv.errors) {
+            throw new Error(`Masked schema for class ${modelName} was not valid!`)
+        }
+        this.fieldsValidators[modelName] = ajv.compile(maskedSchema)
+    }
+
+    /**
+     * Validates specified model filters
+     * @param {string} modelName
+     * @param {JSON} filters
+     * @returns {{isValid: boolean, errors: *[]}}
+     */
+    validateFilters (modelName, filters) {
+        if (!this.models.includes(modelName)) {
+            throw new Error(`Unregistered model name: ${modelName}`)
+        }
+        const validator = this.filterValidators[modelName]
+        const isValid = validator(filters)
+        if (isValid) {
+            return { isValid, errors: [] }
+        } else {
+            return { isValid, errors: validator.errors }
+        }
+    }
+
+    /**
+     * Validates normalized field-string (Subset of model fields in GQL notation)
+     * @param modelName
+     * @param {string} modelName
+     * @param {string} fieldString
+     * @returns {{isValid: boolean, errors: string[]}|{isValid, errors}|{isValid, errors: *[]}}
+     */
+    validateFields (modelName, fieldString) {
+        if (!this.models.includes(modelName)) {
+            throw new Error(`Unregistered model name: ${modelName}`)
+        }
+        const validator = this.fieldsValidators[modelName]
```

Comment:
```suggestion
        const validator = this.fieldsValidators[modelName]
        if (!validator) {
            throw new Error('aaaaaaa')
        }
```

Reply:
Validator will always be here...since it occurs on model registration